File: API/Entities/Loan.cs
using System;

namespace API.Entities;

public class Loan
{
     public int Id { get; set; }
    public int BorrowerId { get; set; }
    public int LoanOfferId { get; set; }
    public decimal PrincipalAmount { get; set; }
    public decimal RemainingBalance { get; set; }
    public decimal InterestRate { get; set; }
    public int DurationInMonths { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public LoanStatus Status { get; set; }
    public virtual LoanOffer LoanOffer { get; set; }
    public virtual User Borrower { get; set; }
    public virtual ICollection<Payment> Payments { get; set; }
}

public enum LoanStatus
    {
        Active,
        Paid,
        Defaulted
    }

File: client/src/app/_models/loan.ts
import { LoanOffer } from "./loan-offer";
import { Payment } from "./payment";
import { User } from "./user";

export interface Loan {
    id: number;
    borrowerId: number;
    loanOfferId: number;
    principalAmount: number;
    remainingBalance: number;
    interestRate: number;
    durationInYears: number;
    monthlyRepayment:number;
    totalRepayment:number;
    startDate: Date;
    endDate: Date;
    status: LoanStatus;
    loanOffer: LoanOffer;
    borrower: User;
    payments: Payment[];
}

export enum LoanStatus {
    Active = 'Active',
    Paid = 'Paid',
    Defaulted = 'Defaulted'
}